### ISSUE TITLE
java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by AdamGrzybkowski** [dsrees/JavaPhoenixClient#69]
I'm getting a lot of this on a recent library version. Could you look at this?

```
java.util.ConcurrentModificationException: 
  at java.util.ArrayList$Itr.next (ArrayList.java:860)
  at org.phoenixframework.Socket.onConnectionError$JavaPhoenixClient (Socket.java:511)
  at org.phoenixframework.Socket$connect$3.invoke (Socket.java:235)
  at org.phoenixframework.Socket$connect$3.invoke (Socket.java:61)
  at org.phoenixframework.WebSocketTransport.onFailure (WebSocketTransport.java:136)
  at okhttp3.internal.ws.RealWebSocket.failWebSocket (RealWebSocket.java:570)
  at okhttp3.internal.ws.RealWebSocket$1.onFailure (RealWebSocket.java:216)
  at okhttp3.RealCall$AsyncCall.execute (RealCall.java:180)
  at okhttp3.internal.NamedRunnable.run (NamedRunnable.java:32)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1162)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:636)
  at java.lang.Thread.run (Thread.java:784)
```

**Comment by dsrees** [comment]
@AdamGrzybkowski  You're on v0.3.1, correct?

**Comment by AdamGrzybkowski** [comment]
@dsrees yup, I'm using my fork but I didn't modify anything related to this. I think this might be a side effect of the recent [fix](https://github.com/dsrees/JavaPhoenixClient/pull/66) as this one crash is not present anymore

**Comment by dsrees** [comment]
I think #67 addresses this issue, I just need to get tests written for it

**Comment by dsrees** [comment]
Resolved in #70. Will be in release v0.3.2
